src/main/java/model/LottoResult.java

```diff
@@ -0,0 +1,33 @@
+package model;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class LottoResult {
+    private static final int INITIAL_COUNT = 0;
+    private static final int COUNT_INCREMENT = 1;
+    private final List<LottoRank> lottoRanks;
+
+    public LottoResult(List<LottoRank> lottoRanks) {
+        this.lottoRanks = lottoRanks;
+    }
+
+    public Map<LottoRank, Integer> calculateStatistics() {
+        Map<LottoRank, Integer> statistics = new HashMap<>();
+        for (LottoRank lottoRank : lottoRanks) {
+            statistics.put(lottoRank, statistics.getOrDefault(lottoRank, INITIAL_COUNT) + COUNT_INCREMENT);
+        }
+        return statistics;
```

Comment:
아래와 같이 변경해도 동일한 결과를 얻을 수 있을 것 같아요. 🙂
```suggestion
        return lottoRanks.stream()
            .collect(Collectors.groupingBy(rank -> rank, Collectors.counting()));
```